File: WhatTheMessenger.Core/Models/Message.cs
namespace WhatTheMessenger.Core.Models;

public enum MessageStatus
{
    Fail,
    Sent,
    Delivered,
    Read
}

public sealed class Message
{
    public Guid Id { get; set; }
    public required Chat Chat { get; set; }
    public User Sender { get; set; } = null!;
    public Guid SenderId { get; set; }
    public required string Content { get; set; }
    public required DateTime SentAt { get; set; }
    public required MessageStatus Status { get; set; }
}

File: client/src/app/core/models/message.ts
import { guid } from "../../primitives";

export interface Message {
    id: guid;
    chatId: guid;
    senderId: guid;
    content: string;
    sentAt: Date;
    status: MessageStatus;
}

export enum MessageStatus {
    Fail,
    Sent,
    Delivered,
    Read
}